File: javascript/high-scores/high-scores.js
export class HighScores {

    #scoresList;

    constructor(scores) {
        this.#scoresList = [...scores];
    }

    get scores() {
        return this.#scoresList;
    }

    get latest() {
        return this.#scoresList[this.#scoresList.length - 1];
    }

    get personalBest() {
        return Math.max(...this.#scoresList);
    }

    get personalTopThree() {
        return this.#scoresList.toSorted((a, b) => b - a).slice(0, 3);
    }
}

File: java/high-scores/src/main/java/HighScores.java
import java.util.Comparator;
import java.util.List;

class HighScores {

    private final List<Integer> scores;

    public HighScores(List<Integer> highScores) {
        this.scores = highScores;
    }

    List<Integer> scores() {
        return this.scores;
    }

    Integer latest() {
        return this.scores.getLast();
    }

    Integer personalBest() {
        return this.scores.stream().max(Integer::compareTo).orElseThrow();
    }

    List<Integer> personalTopThree() {
        return this.scores.stream().sorted(Comparator.reverseOrder()).limit(3).toList();
    }

}
